locations/method_locations_test.go

```diff
@@ -117,3 +117,33 @@ func TestMethodSignature(t *testing.T) {
 		}
 	}
 }
+
+func TestMethodOption(t *testing.T) {
+	f := parse(t, `
+	  import "google/api/client.proto";
+		service Library {
+		  rpc GetBook(GetBookRequest) returns (Book) {
+		    option deadline = 60.0;
+		  }
+		  rpc WriteBook(NewBook) returns (Book) {}
+		}
+		message GetBookRequest{}
+		message Book {}
+		message NewBook {}
+	`)
+
+	for _, test := range []struct {
+		name      string
+		methodIdx int
+		option    string
+		want      []int32
+	}{
+		{"First", 0, "deadline", []int32{5, 20, 43}},
+		{"Second", 1, "deadline", nil},
+	} {
+		loc := MethodOption(f.GetServices()[0].GetMethods()[test.methodIdx], test.option)
```

Comment:
imo, we should be using, https://pkg.go.dev/testing#T.Run.

e.g.,

```
func TestFoo(t *testing.T) {
    for _, test  := range []struct {
        name string
        ...
    }{
        {...},
        {...},
    } {
        t.Run(test.name, func (t *testing.T) {
            ...
            if actual != expected {
                t.Errorf("Expected, %s, was %s", expected, actual)
            }
        }
    }
}
```

Reply:
In turn, `name` will appear in output to stdout when running tests and could optionally be omitted from the `Fatalf` or `Errorf` invocation. Finally, this is the recommended table test method from the golang blog, "[Go 1.7 also introduces a Run method for creating subtests](https://go.dev/blog/subtests)".